src/pages/index.js

```diff
@@ -0,0 +1,329 @@
+        previewModalCaptionElement.textContent = data.name;
+    });
+
+    cardLikeButton.addEventListener("click", (evt) => handleLikeCard(evt, data._id, data.isLiked));
+
+    cardDeleteButton.addEventListener("click", () => handleDeleteCard(cardElement, data._id));
+
+    return cardElement;
+
+}
+
+
+function openModal (modal) {
+    modal.classList.add("modal_opened");
+    document.addEventListener("keydown", handleEscape);
+}
+
+function closeModal (modal) {
+    modal.classList.remove("modal_opened");
+    document.removeEventListener("keydown", handleEscape);
+}
+
+function handleLikeCard(event, cardId, isLiked) {
+  api.updateLikeCard(cardId, isLiked)
+    .then(() => {
+      event.target.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
+function handleProfileFormSubmit(evt) {
+    evt.preventDefault();
+
+    const submitButton = evt.submitter;
+    setButtonText(submitButton, true, "Saving...");
+
+    api.updateUserInfo({name: editProfileNameInput.value, about: editProfileDescriptionElement.value})
+    .then((userData) => {
+        profileNameElement.textContent = userData.name;
+        profileDescriptionElement.textContent = userData.about;
+        closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, true, "Saving...", "Save");
+    });
+
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const cardSubmitButton = evt.submitter;
+  cardSubmitButton.disabled = true;
+  setButtonText(cardSubmitButton, "Saving...");
+
+  const inputValue = { 
+      name: cardCaptionInput.value, 
+      link: cardLinkInput.value
+  };
+
+  api.createCard(inputValue)
+      .then((newCard) => {
+          const cardEl = getCardElement(newCard);
+          cardList.prepend(cardEl);
+          closeModal(cardModal);
+          cardFormElement.reset();
+      })
+      .catch(console.error)
+      .finally(() => {
+          cardSubmitButton.disabled = false;
+          setButtonText(cardSubmitButton, "Save");
+      });
+}
+
+document.getElementById("edit-avatar-form").addEventListener("submit", handleAvatarFormSubmit);
+
+
+function handleAvatarFormSubmit(evt) {
+    evt.preventDefault();
+
+
+    const avatarLinkInput = avatarModal.querySelector("#profile-avatar-input");
+
+    const avatarSubmitButton = evt.submitter;
+    setButtonText(avatarSubmitButton, true, "Saving...");
+    avatarSubmitButton.disabled = true;
+
+    api.updateUserAvatar(avatarLinkInput.value)
+    .then((userData) => {
+        spotsAvatar.src = userData.avatar;
+        closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+        avatarSubmitButton.disabled = false;
+        setButtonText(avatarSubmitButton, false, "Save", "Saving...");
+    });
+}
+
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true, "Delete");
```

Comment:
Please, change the order of the texts because the 3rd argument is the loading text, and the 4ht is the default on



```js
  setButtonText(submitButton, true, "Deleting...");

```